src/main/scala/mesosphere/marathon/core/task/termination/impl/TaskKillServiceActor.scala

```diff
@@ -0,0 +1,181 @@
+package mesosphere.marathon.core.task.termination.impl
+
+import akka.actor.{ Actor, ActorLogging, Cancellable, Props }
+import mesosphere.marathon.MarathonSchedulerDriverHolder
+import mesosphere.marathon.core.base.Clock
+import mesosphere.marathon.core.task.termination.TaskKillConfig
+import mesosphere.marathon.state.Timestamp
+import mesosphere.marathon.core.task.{ Task, TaskStateOp }
+import mesosphere.marathon.core.task.tracker.{ TaskStateOpProcessor, TaskTracker }
+import mesosphere.marathon.core.event.MesosStatusUpdateEvent
+
+import scala.collection.mutable
+import scala.concurrent.Promise
+
+/**
+  * An actor that handles killing tasks in chunks and depending on the task state.
+  * Lost tasks will simply be expunged from state, while active tasks will be killed
+  * via the scheduler driver. There is be a maximum number of kills in flight, and
+  * the service will only issue more kills when tasks are reported terminal.
+  */
+private[impl] class TaskKillServiceActor(
```

Comment:
a) the current implementation of the actor has merely one state, so that might be considered too much. Yet I like the concise and readable notation of akka FSM, so yes, I consider that if others are fine with that decision.

b) Makes sense to me. I was thinking about doing it it the beginning but chose a less sophisticated approach to get the ball rolling and provide the main functionality. When batch processing the messages, an FSM would make sense, so doing b) justifies a) ;)
OTOH, batch processing would slightly defer incoming kills – in a range of hundreds of milliseconds, that would be just fine I guess.